### ISSUE TITLE
Project status (fork more current/usable)

### ISSUE THREAD

**Issue by alerque** [rawstudio/rawstudio#53]
For others that come looking here, it looks like the most current usable fork of this project is currently [sergiomb2's fork](https://github.com/sergiomb2/rawstudio). Fedora has started picking up packages from there. Bizarrely they've opted to call it 2.1 (even though 2.0 never saw more than RC tags here and neither fork has ever tagged a 2.1).

I'm going to be porting the Arch Linux packages over to that fork shortly.

If anybody hears more recent project status information (signs of life from original developers, etc.) I'd be happy to hear about it, and notifications on this issue might be a good place for other developers to track this.

I'd love to see some distro maintainers and active fork people given access to this org & the main repository so we could at least keep a stable release in a compilable state!

**Comment by guzzisti** [comment]
By "Arch Linux packages" you mean the AUR package https://aur.archlinux.org/packages/rawstudio-git? 
This doesn't build for me currently, maybe you can have a look:

```
==> Starting pkgver()...
fatal: No names found, cannot describe anything.
==> ERROR: pkgver is not allowed to be empty.
==> ERROR: pkgver() generated an invalid version: 
```

**Comment by sergiomb2** [comment]
committer/maintainer on Fedora Packager is (also) me .

**Comment by alerque** [comment]
Thanks for the report @guzzisti. This seems to be something that happened since I updated the package yesterday because what worked for me then does not now. It seems the upstream repository from @sergiomb2 dropped all of the git tags from the repo, hence the AUR package can no longer make a guess at what the last version + number of commits since then is. Can you push at least the set of tags as found on _this_ repository to your fork as well? They are kind of needed to keep the fork building as expected.

**Comment by alerque** [comment]
You should be able to compile `rawstudio-git` now @guzzisti. I added a temporary hack to tag b548ed4 in the cloned git source as v2.0 so that the version calculator can orient itself until the forked repository gains its tags back.

**Comment by alerque** [comment]
Pinging @abrander and @anderskvist, as the two listed people in this Github org are either of you around to comment on this situation? It would be really nice if you were willing to open this org up a little bit to one or two more maintainers. Initially just addressing the build, release numbering, and packing issues so that distros that are still trying to carry this don't have to drop in and don't end up in a crazy mess of versions not actually being what they say on the tin. Secondarily a few active maintainers might spark interest in contributions again and maybe move development along just a bit. What do you say?

**Comment by abrander** [comment]
First of a little history. The Rawstudio project was started more than 12 years ago. A lot has happened in life since. Children were born, jobs were changed. Hobbies were changed, and life moved on faster and faster.

I still use Rawstudio from time to time, and I still prefer it over other raw converters, but I must admit that bringing Rawstudio back into shape is a task that I simply don't have the energy for at the moment.

I think it's a fabulous idea to add one or two maintainers with the needed energy and expertise.

**Comment by alerque** [comment]
Thanks for responding @abrander! And believe me I've been involved in OSS projects long enough to know how it goes. I've been the one that's dropped out for years at a time too, or changed interest and switched projects.

Quite honestly I don't have the time or drive to push this project forward as far as really bringing it back into shape (e.g. updating the UI toolkits, etc.) but I do use it and keep it packaged it for Arch Linux — and I'd like to see it continue being _usable_ for folks even if it doesn't get an overhaul any time soon.

I would propose starting out by adding myself and @sergiomb2 as maintainers to the org. He's done some of the patching necessary to make this build on some distros already in his fork and I'd like to see that work brought back here. We both work on making this functional in our respective distros (Fedora is his case and Arch Linux in mine). Neither of us may be the ones to drive major development work, but if we had access to work on issues here, review some of the outstanding PRs, and generally keep from "broken window" syndrome from setting in that would be a good start.

Hopefully that little bit of cleanup and a release cycle so the project doesn't look so in limbo might encourage some other expertise to consider contributing in the future.

**Comment by sergiomb2** [comment]
@abrander it will be an honor add my pull requests in this repo , we know all limitations of rawstudio, gtk3 is not complete stable , crash on wayland , ufraw just load old cameras lens. but ATM project is still alive, and if is came up someone to hack the code and fix the issues it would be great , I also handn't much time even to use rawstudio itself. 
I end up also keep ufraw alive, the upgrade to ufraw 0.23 was done by me, the future maybe is swap to libRaw . 
My major concern it is keep things organized the best it is possible , picking patches that appears over the internet

**Comment by anderskvist** [comment]
Big :+1: for giving others access to keep Rawstudio alive! Would be so great to see what we've started being adopted and maintained by others :smiley: Thanks for wanting this! Just as @abrander, there just isn't the time that we had 10+ years ago where we first sat in a bar with a sketchbook and made the first UI drawings and later sat together very often and coded day and nights and ate a lot of cheap frozen pizzas :joy: But this project gave us a LOT back - we attended 3 Libre Graphics Meeting to meet up with like minded and made a lot of friends - oh I miss those times, but life have to move on...

**Comment by alerque** [comment]
Okay, so it seems both of you who have repository access now are in favor of what I proposed. Who has to actually poke the buttons to grant access?

**Comment by alerque** [comment]
## End of Year Opinion Poll¹

Which is most likely to happen in 2020?

(Place your bets using the corresponding reaction emoji on this post.)

* :rocket: = A private US company takes humans back to the Moon.
* :eyes: = China lands an operational rover on Mars.
* :tada: = Adobe open sources their entire lineup of graphics software and contributors promptly port Photoshop to run on Linux; the UI ironically uses GTK, the GIMP Toolkit.
* :smile: = An OSS project's original developers that have already agreed to open access so  willing contributors can continue maintenance actually flips the switch so said new maintainers have access.

¹ <sub>One of these is much less likely to be completed in 2020 than the others.</sub>

**Comment by abrander** [comment]
@alerque I appreciate your humorous attempt to bring attention to this issue 😄 

Adding new maintainers sounds like the right thing to do, but I'm somewhat curious about what's stopping any contributions right now.

> but if we had access to work on issues here, review some of the outstanding PRs, and generally keep from "broken window" syndrome from setting in that would be a good start.

Is there anything stopping this right now?

> Hopefully that little bit of cleanup and a release cycle so the project doesn't look so in limbo might encourage some other expertise to consider contributing in the future.

That would be awesome!

I propose someone (you?) open an issue like "Bring Rawstudio back on track" where we can track what should be done. One of the things that should be done could be adding new maintainers - but how about we try to do something together? If that works out, we could formalize everything and flip the switch.

**Comment by alerque** [comment]
> I'm somewhat curious about what's stopping any contributions right now.

Lack of responsiveness from maintainers is stopping contributions. No seriously, commits that fix known problems have sat around in pull requests for _YEARS_ without feedback from any active developer. That is completely demotivating to any potential contributor. There is only marginal indication that the PRs have even been seen, there is no feedback on whether the commits are good, bad, need testing, whatever. Without responsive maintainers there is no path for contributors to follow.

Furthermore it's gotten to the point where the official released version doesn't run on most distros, and with 500+ commits since the last release and the only hope of it working, distro packagers (like myself) have had to resort to forking just to ship something that sort of works.

> Is there anything stopping this right now?

Yes. Not having responsive maintainers has stopped and is stopping contributions.

> One of the things that should be done could be adding new maintainers - but how about we try to do something together?

I'm not proposing kicking you out of the project, but having active maintainers is not something that "could be" done, it is _the single thing that needs to happen first_. Without that it's going to be pretty much impossible to rally support, get feedback, motivate contributors to cleanup their contributions so they can be merged, or any of that.

You manifestly don't have the time or interest right now and haven't for years. That isn't an indictment of wrongdoing, it's just the way things go. I completely understand that, but please at least realize how much having unresponsive maintainers is a barrier to contribution.

Right now my proposal is simple. Add me to the org with enough permissions to manage this repo. I'll work through the issue tracker and PRs, particularly with @sergiomb2 and help him get the current distro status-quo moved from patches to the core project so what is being shipped is this project, not a hodge-podge of our forks and patches.

After that and as interest arises from contributors we can talk about what might go into future work, possibly work up a development roadmap. That can all be done here on Github in issues and the "we" of "we can do something together" can of course include the original developers, but only is so far as they are active an jump into things.

**Comment by guzzisti** [comment]
Well, i don't think we'll see any progress on this repo anytime soon.

Maybe we could try something different:
There seems to be a few people around who'd like to maintain atleast the most basic stuff. What do you think about these people are creating a new repo where they can contribute to? Just to have something that builds and could be used on most distros.
@sergiomb2 s fork may be a good starting point.

**Comment by alerque** [comment]
@abrander and @anderskvist you are the only two members currently listed publically on this org. At this point we don't really have anybody else to turn to. Both of you have agreed in comments above that opening access to another maintainer(s) is the right thing do do. Nobody has objected to my proposal or suggested I would ruin things if I take the steps I suggested.

Whose call is this to make and who can actually flip the switch?

**Comment by sergiomb2** [comment]
Hello, 
I just want to say that be maintainer of this project is different of have a fork, the responsibility is different , we will have more users to test my commits , etc . 
Anyway https://github.com/sergiomb2/rawstudio/issues/1#issuecomment-542639817 , updates of ICC-profiles or lcms may have introduce problems , that aren't solved ...

**Comment by abrander** [comment]
@alerque @sergiomb2 @anderskvist 

Thanks for your patience. To be honest I would have no problem adding anybody who had been a long time contributor as a maintainer. But in this case, there's not much history of collaboration in this project - probably because the lack of active maintainers. That doesn't mean it's a bad idea, but it doesn't make the decision any easier.

Would any of you be interested in scheduling a conference call where we can discuss this further? We're all in agreement that something must be done. Maybe we could discuss exactly what and how?

**Comment by alerque** [comment]
@abrander Sure. When? Are you on Signal or Whatsapp? If so +905069078206, otherwise let me know what you'd like to use. I have a US number I can access if need be.

I understand the cart/horse problem here, but given the stagnated state of the project there _are no long time contributors_. My personal interest for now is just in getting all the build issues fixes so it can be packaged in Linux distros. Currently it is so broken distros are dropping it! Once the build issue are fixed and downstream patches are integrated in such a way that they work across distros, then setting up some CI stuff so any contributions could be verified in a reasonable amount of time would be my next step, then working through any other PRs / issues and responding to things might drum up some interest in fixing or adding features. I don't have any intentions of a big development push, I would just help steer contributors as they came along.

**Comment by abrander** [comment]
@alerque My plan was to set up an old school regular telephone conference call, but maybe we can simply discuss it here. That was the reason behind my request for a "Bring Rawstudio back on track" issue as a place to discuss a plan.

You mentioned a few other things:

> getting all the build issues fixed

You're more than welcome to open pull requests if you have anything laying around.

> CI stuff so any contributions could be verified in a reasonable amount of time

And here is the heart of the problem. It's pretty easy to verify that Rawstudio builds. What gets complicated is checking that it actually works. A lot can be tested by CI, but a lot cannot, and will require visual inspection of Rawstudio's output *and* preview. That's the problem with #37, #38 and #52.
Rawstudio's repository of test images is huge. It's very time consuming to verify conversion, and subtle code and algorithm changes can result in dramatic regressions. One could argue that it's useless to have a raw converter that works if it doesn't build thou :smirk: 

> help steer contributors

That would be awesome! I can even see myself as a contributor.

**Comment by sergiomb2** [comment]
I wrote somewhere, we should migrate to LibRaw [1] is based on the source codes of the dcraw utility, ufraw and dcraw seems that haven't updates since 2017 .
I will try review my PRs #37, #38 and #52  


[1] 
https://github.com/LibRaw/LibRaw
https://www.libraw.org/

**Comment by abrander** [comment]
> I wrote somewhere, we should migrate to LibRaw [1] is based on the source codes of the dcraw utility, ufraw and dcraw seems that haven't updates since 2017 .

I think it's a fabulous idea porting Rawstudio to LibRaw a sa replacement for dcraw while still using RawSpeed as a first choice in decoder.

**Comment by abrander** [comment]
> I can even see myself as a contributor.

Well. That happened. For anyone interested in reviewing pull requests, here are a few easy frags. It's mostly code removal.

Code reviews and comments welcome.

#54: Require LCMS2
#55: Remove custom GTK+ theme
#56: Add build instructions
#57: Remove support for Windows and macOS

**Comment by sergiomb2** [comment]
all 4 LGTM , I haven't check the 4th patch exhaustively

**Comment by raw-thorn** [comment]
I'm using latest version since September 15 on Gentoo Linux (using mostly stable programs).
I works fine for me.
Thanks for updating rawstudio.

**Comment by AlvesR0** [comment]
> 
> 
> @alerque @sergiomb2 @anderskvist
> 
> Thanks for your patience. To be honest I would have no problem adding anybody who had been a long time contributor as a maintainer. But in this case, there's not much history of collaboration in this project - probably because the lack of active maintainers. That doesn't mean it's a bad idea, but it doesn't make the decision any easier.
> 
> Would any of you be interested in scheduling a conference call where we can discuss this further? We're all in agreement that something must be done. Maybe we could discuss exactly what and how?

I practically found 3 videos (official website) and no information on social networks about RawStudio (ex: youtube, twitter, discord, etc ...) Many users are unaware of its existence. : (
I will leave some suggestions. 1- create a sub-forum on pixls.us. Certainly the project will have support. 2- Donations through: Vakinha, Patreon, Apoia-se, etc ... 
3- Create a group on the telegram. In order to bring people who want to make RawStudio the best there is when it comes to post-production tools for photography.

**Comment by esbante** [comment]
Rawstudio seems to work ok on ubuntu currently, on manjaro there is a problem with the vignette tool and some dng:s, as well as overexposed areas (in dng:s), but overall it is usable now. But wouldn't one way to make it usable on all distributions, with (hopefully) no need to revise the code in the future, be to make an appimage from the last version that was working without issues - Rawstudio-daily from the rawstudio ppa of ubuntu 14.04? For those who wish to use more recent cameras there is always the option to convert raw files to dng:s. With the current version from this repository that is not the best option, since there is a problem with dngs. What do you think?

**Comment by sergiomb2** [comment]
we need migrate from GConf2 to gsetting (I guess) and integrate libRaw .
GConf2 will be remove from distros soon or later, Fedora is planing it for 2022 or 2023 .
About  libRaw, is continuation of ufraw etc , but with the good thing of be a library with api , so we can build with it as an external dependency .

**Comment by esbante** [comment]
That would be great, if possible. I will gladly contribute with testing, but am not able to do programming unfortunately.
